### DISCUSSION TITLE
NFS to existing NAS?

### DISCUSSION THREAD

**Discussion by ChaOConnor** [retronas/discussions#1] [Q&A] [answered]
The Cockpit UI offers and easy was to map an existing NFS share to a folder on retronas.  For those using something like unRaid, would this be a good option in the near term?

You could spin up a VM or even use the RPi4 and instead of adding a USB drive, create your TLD to be the NFS Share on unRaid you've exported, called "retronas".  The beauty in this, just as with a USB drive, it would all be "portable" so eventually if Docker, pre-configured VMs, or RPi4 uses evolve, your data set remains the same.

Furthermore, listening to Bob and Dan's discussion on RetroRGB, the number 1 item I'm most interested in and exited about, is the ability to integrate and automate checksum of game files.  Right now I've got directories all over unRaid for backup purposes but are duplicating space.  retronas takes care of a lot of that due to it's use of symlinks (and in truth I could do more of that natively with unRaid but retronas makes it so easy), but also which files are definitive?  Do I have the right one or is it a corrupt download?  The ability to integrate lists into retronas and help create bonafide sets would be a game changer.  

Example: Point retronas to a directory on the USB Drive or unRaid Share and let it be "managed" by retronas.  One of the options/scripts would be to use Smokemoster's scripts on that directory to create a curated set of correct game files against known releases (Smokemoster's lists, No-Intro, etc.).  

Lots of thoughts here, just throwing it out for discussion.  Love this project, it's making me want to learn Ansible so I can help out!  Thank you!

**Comment by danmons** [comment, marked as answer]
Yes, this should work on any network file system that supports symlilnks, which NFS does.

So if all you wanted was the SMB component, and your current NAS offered that, as above:

1) spin up a VM, install RetroNAS
2) mount your underlying NAS via NFS and tell RetroNAS that's your top level dir
3) run any of the tools that generate the symlinks and directories you need
4) spin down the VM
5) Share out the relevant parts of the file system from your NAS tool

You won't get all the bespoke protocols (AppleTalk, EtherDFS, NFSv2, ps3netsrv, and upcoming DECnet/FSP/etc), and potentially not have SMB1 depending on your make/model of NAS, but you'll at least get the symlink structure that you can share out to different things manually.
